File: 704.binary-search.c
/*
 * @lc app=leetcode id=704 lang=c
 *
 * [704] Binary Search
 */

// @lc code=start
int search(int* nums, int numsSize, int target) {
    
    //set initial values
    int low = 0;
    int high = numsSize - 1;
    int mid = (low + high) / 2;
    int guess = nums[mid];

    //loop until value found or constraints met
    while (low <= high) {
        mid = (low + high) / 2;
        guess = nums[mid];

        if(guess == target) {
            return mid;
        }
        else if (guess > target) {
            high = mid - 1;
        }
        else {
            low = mid + 1;
        }
    }

    //will only return -1 if target not found
    return -1;

}
// @lc code=end

File: 704.binary-search.go
/*
 * @lc app=leetcode id=704 lang=golang
 *
 * [704] Binary Search
 */

// @lc code=start
func search(nums []int, target int) int {
    
    low := 0
    high := len(nums) - 1

	for(low <= high) {

		middle := (low + high) / 2;
		guess := nums[middle];

		if guess > target {
			high = middle - 1;
		} else if guess < target {
			low = middle + 1;
		} else {
			return middle;
		}
	}

	//shouldn't reach here
	return -1;
}
}
// @lc code=end

